### ISSUE TITLE
Add custom ML model

### ISSUE THREAD

**Issue by AlexanderGbd** [ecomontec/ecoSound-web#562]
We would like to integrate our customly trained ML model for insect detection at some point, when it is in a good state. The model builds on python and is based on the autrainer library (https://autrainer.github.io/autrainer/index.html). We also plan on uploading the model on huggingface to make it easily updateable and usable. To run the model, a Python environment in which autrainer can be installed and executed should therefore be sufficient.

**Comment by kdarras** [comment]
A general Python-based framework for integrating models will be needed for the re-factoring and new architecture. Huggingface could be used to retrieve models but this needs to be investigated further.

**Comment by mcauchoix** [comment]
Here is the last model version https://huggingface.co/AlexanderGbd/insects-base-cnn10-96k-t

**Comment by AlexanderGbd** [comment]
There are two ways to use the model:

### 1. Full audio inference

You can apply the model directly to the full audio files. This will generate a `results.csv` where each row corresponds to one file, with the following format:

filename, prediction, output, target_class1, target_class2, target_class3, ...

- `filename`: Name of the audio file  
- `prediction`: Predicted target classes  
- `output`: Model output (e.g., raw or processed logits/scores)  
- `target_classX`: Confidence scores for each class  

### 2. Sliding window inference (recommended)

In this approach, the model is applied to short segments/windows of the audio. The output csv contains one row per segment, with the following format:

filename, offset, prediction, output, target_class1, target_class2, target_class3, ...

- `offset`: Start and end time of the segment (e.g., `0.0-4.0` or `4.0-8.0`)  
- `prediction`, `output`, and class scores are as described above  

After all segments of an audio file have been processed, an additional row is included in the csv that represents a **majority vote** over all segment predictions. This aggregated result lists target classes that appear in at least half of the segments. You can choose to use this majority vote or rely on the individual segment predictions. If you're only interested in per-segment results, you can simply ignore or filter out these rows during post-processing.

> **Note:** Although the model on Hugging Face is a simple baseline, the output format will remain the same in the updated version.

**Comment by kdarras** [comment]
Hi @mcauchoix we will follow the usual testing and deployment procedure. The German VM is not for testing, it is a production environment and I will share access credentials with Elodie and David to our French test VM once the functionality is ready for testing.

Indeed implementing the recommended approach with the sliding window ("2" above) will be best but we will discard the majority vote as this cannot be integrated as tags. Everything else, apart from some re-casting of the results CSV, is the same as BirdNET.

The crucial point is to implement this functionality in a truly sustainable way through the Huggingface API connection, as there will be more models to implement from Huggingface. I will discuss the feasibility with Dilong.

**Comment by LiuDilongNJ** [comment]
Could you kindly send me the problematic recording and its parameters?
